source-code/sdk-js/src/runtime/runtime.ts

```diff
@@ -0,0 +1,53 @@
+import type { Resource } from "@inlang/core/ast"
+import { BaseLookupFunctionArgs, createLookupFunction, LookupFunction } from "./lookup-function.js"
+
+const dummyResource = {
+	type: "Resource",
+	languageTag: { type: "LanguageTag", name: "en" },
+	body: [],
+} satisfies Resource
+
+const fallbackLookupFunction: LookupFunction = () => ""
+
+export type RuntimeState<Language extends string = string> = {
+	resources: Map<Language, Resource>
+	language: Language
+}
+
+export const initRuntime = <
+	Language extends string,
+	LookupFunctionArgs extends BaseLookupFunctionArgs = BaseLookupFunctionArgs,
+>() => initBaseRuntime<Language, LookupFunctionArgs>()
+
+export const initBaseRuntime = <
+	Language extends string,
+	LookupFunctionArgs extends BaseLookupFunctionArgs = BaseLookupFunctionArgs,
+>(
+	state: RuntimeState<Language> = {
+		resources: new Map(),
+		language: undefined as unknown as Language,
```

Comment:
@samuelstroschein @benjaminpreiss what do you thing here?

Reply:
If the default value is `undefined` now, we can always change the value later on.

I assume that real-world usage will show what a good default value is. If we try to set a value now and realize that `undefined` is better, changing this would likely be a breaking change.

Reply:
But what should the type definition of `getLanguage` be? `string | undefined`?

Reply:
`string` there will always be a fallback, correct?

Reply:
Not in the core library. If the language was never set, then we don't have a value. We should probably use `steing | undefined`